### PR TITLE
Merge master HEAD into openj9-staging

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Merge master HEAD into openj9-staging

Conflicts:
```
	test/jdk/java/foreign/NativeTestHelper.java
```
Modified
```
	src/java.base/share/classes/jdk/internal/foreign/abi/ppc64/aix/AixPPC64Linker.java
```

Depends on
* https://github.com/eclipse-openj9/openj9/pull/18296

FYI @ChengJin01 

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

**Comment by keithc-ca** [comment]
`openjdk-tag.gmk` should be updated for the new tag.

File: src/java.base/share/classes/jdk/internal/foreign/abi/ppc64/aix/AixPPC64Linker.java [hunk 1]
```diff
@@ -48,6 +52,9 @@
  */
 public final class AixPPC64Linker extends AbstractLinker {
 
+    static final Map<String, MemoryLayout> CANONICAL_LAYOUTS =
+            SharedUtils.canonicalLayouts(ValueLayout.JAVA_LONG, ValueLayout.JAVA_LONG, ValueLayout.JAVA_INT);
```
Comment (keithc-ca) [inline src/java.base/share/classes/jdk/internal/foreign/abi/ppc64/aix/AixPPC64Linker.java:56]: This field should be `private` and the entry for `double` needs a different alignment:
```java
    private static final Map<String, MemoryLayout> CANONICAL_LAYOUTS;

    static {
        Map<String, MemoryLayout> layouts = new HashMap(
                SharedUtils.canonicalLayouts(
                        ValueLayout.JAVA_LONG,
                        ValueLayout.JAVA_LONG,
                        ValueLayout.JAVA_INT));
        layouts.put("double", ValueLayout.JAVA_DOUBLE.withByteAlignment(4));
        CANONICAL_LAYOUTS = Map.copyOf(layouts);
	}
```

File: test/jdk/java/foreign/NativeTestHelper.java [hunk 2]
```diff
@@ -93,44 +93,63 @@ public static boolean isPointer(MemoryLayout layout) {
...
     /**
      * The layout for the {@code float} C type
      */
-    public static final ValueLayout.OfFloat C_FLOAT = ValueLayout.JAVA_FLOAT;
+    public static final ValueLayout.OfFloat C_FLOAT = (ValueLayout.OfFloat) LINKER.canonicalLayouts().get("float");
     /**
      * The layout for the {@code double} C type
      */
-    public static final ValueLayout.OfDouble C_DOUBLE = ValueLayout.JAVA_DOUBLE.withByteAlignment(isAixOS ? 4 : Double.BYTES);
+    public static final ValueLayout.OfDouble C_DOUBLE = isAixOS ? ValueLayout.JAVA_DOUBLE.withByteAlignment(4) : (ValueLayout.OfDouble) LINKER.canonicalLayouts().get("double");
```
Comment (keithc-ca) [inline test/jdk/java/foreign/NativeTestHelper.java:128]: This file can be identical to upstream with my suggested change to `AixPPC64Linker`.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: src/java.base/share/classes/jdk/internal/foreign/abi/ppc64/aix/AixPPC64Linker.java (hunk 1 above)
Comment (JasonFengJ9) [reply src/java.base/share/classes/jdk/internal/foreign/abi/ppc64/aix/AixPPC64Linker.java]: Changed as per the suggestion.

File: test/jdk/java/foreign/NativeTestHelper.java (hunk 2 above)
Comment (JasonFengJ9) [reply test/jdk/java/foreign/NativeTestHelper.java]: Restored upstream version.

**Review by keithc-ca — APPROVED** [review]

**Comment by pshipton** [comment]
It's only staging, which was broken anyway, not really a concern.

**Pull request merged**